### DISCUSSION TITLE
How to solve when using configpart.withInstanceAttributeOverride() on array field, the new Node will be added to items Node

### DISCUSSION THREAD

**Discussion by manchuanxingmengyaxinhe** [victools/jsonschema-generator#270] [Q&A] [answered]
```java
configPart.withInstanceAttributeOverride(((objectNode, fieldScope, context) -> {
        final JsonSchemaProperty arraySchema = fieldScope.getAnnotationConsideringFieldAndGetter(JsonSchemaProperty.class);
        if (arraySchema != null) {
            if (arraySchema.prefixItems().length != ZERO) {
                PrefixItems[] items = arraySchema.prefixItems();
                final ArrayNode node = objectNode.withArray(MoyeSchemaKeyword.PREFIX_ITEMS);
                for (PrefixItems item : items) {
                    final String name = item.name();
                    final String value = item.value();
                    ObjectNode itemNode = createObjectNode();
                    itemNode.put(name, value);
                    node.add(itemNode);
                }
                objectNode.put(MoyeSchemaKeyword.ITEMS, "false");
            }
            if (arraySchema.maxContains() != -1) {
                objectNode.put(MoyeSchemaKeyword.MAX_CONTAINS, arraySchema.maxContains());
            }
            if (arraySchema.minContains() != -1) {
                objectNode.put(MoyeSchemaKeyword.MIN_CONTAINS, arraySchema.minContains());
            }
        }
}))
```
the schema is shown below. the "prefixItems" is added "items" node
```json
{
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "type": "object",
    "properties": {
        "oneTest": {
            "prefixItems": [
                {
                    "type": "String.class"
                },
                {
                    "enum": "NW, NE, SW, SE"
                }
            ],
            "items": {
                "type": "integer",
                "prefixItems": [
                    {
                        "type": "String.class"
                    },
                    {
                        "enum": "NW, NE, SW, SE"
                    }
                ],
                "items": "false"
            },
            "type": "array"
        }
    }
}
```
thanks for your help

**Comment by CarstenWickner** [comment, marked as answer]
Hi @manchuanxingmengyaxinhe,

The corresponding part from the documentation is this (point 3 in the yellow box):
https://victools.github.io/jsonschema-generator/#generator-individual-configurations
> You may want to specifically consider or return `null` for cases where `FieldScope.isFakeContainerItemScope()`/`MethodScope.isFakeContainerItemScope()` returns `true`.
Because each individual configuration added via `.forField()` or `.forMethods()` is being called upon twice if the encountered type is a "container" (i.e. an array or subtype of `Collection`).


The schema generation will be triggered twice on a field/method with a collection type.
1. The first time, the `FieldScope` in your case would return:
    - `getType()` = `List<TupleValue>` (or whatever list element type you have)
    - `isFakeContainerItemScope()` = `false`
2. The second time, the `FieldScope` in your case would return:
    - `getType()` = `TupleValue` (or whatever list element type you have)
    - `isFakeContainerItemScope()` = `true` ❗

In your case, you should only consider your annotation's `arraySchema` when `isFakeContainerItemScope()` = `false`, but skip it when `true` is returned.

----

Further notes to your example:
- You currently can't override the `items` keyword via an instance attribute override, only a type attribute override or custom definition. Adding `"false"` as text instead of as `boolean` (without surrounding quote symbols) to the `"items"` keyword would be invalid by the way. Maybe a custom definition provider would be the more appropriate choice here, to get full control over this special array handling.
- The `minItems`/`maxItems` keywords, you may want to handle outside of this attribute override, and instead use the "standard configuration": https://victools.github.io/jsonschema-generator/#quot-minitems-quot-keyword
Either way is acceptable of course.

**Reply by manchuanxingmengyaxinhe** [reply]
thanks !
